spline_background.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+from scipy import interpolate
+
+
+def bspline_background(variable, x_values, dim=None):
```

Comment:
With `scipp` we can avoid the need to juggle multiple parameters. Instead of `variable` and `x_values`, accepts a `DataArray` as input, then you can get the x-values as `variable.coords[dim]`.